flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobClient.java

```diff
@@ -551,33 +425,17 @@ else if (response == RETURN_ERROR) {
 	 * @param jobID
 	 *        the ID of job the BLOB belongs to or <code>null</code> to indicate the upload of a
 	 *        content-addressable BLOB
-	 * @param key
-	 *        the key of the BLOB to upload or <code>null</code> to indicate the upload of a content-addressable BLOB
 	 * @throws IOException
 	 *         thrown if an I/O error occurs while writing the header data to the output stream
 	 */
-	private void sendPutHeader(OutputStream outputStream, JobID jobID, String key) throws IOException {
-		// sanity check that either both are null or both are not null
-		if ((jobID != null || key != null) && !(jobID != null && key != null)) {
-			throw new IllegalArgumentException();
-		}
+	private void sendPutHeader(OutputStream outputStream, JobID jobID) throws IOException {
+		checkArgument(jobID == null);
```

Comment:
no - in this PR, the jobID is not used anymore - a future PR will add uses for that again